libcontainer/cgroups/systemd/v2.go

```diff
@@ -290,15 +291,27 @@ func (m *unifiedManager) Apply(pid int) error {
 		return err
 	}
 
-	// When systemd creates a unit owned by a user other than the
-	// user running the service manager (usually root), it chowns
-	// the cgroup directory and the three files mentioned below.
-	// This enables the service to create new scopes/slices and set
-	// the cgroup for its child processes/threads.
-	//
-	// We follow systemd's lead and chown only those same files.
-	filesToChown := []string{".", "cgroup.procs", "cgroup.subtree_control", "cgroup.threads"}
 	if c.OwnerUID != nil {
+		// The kernel exposes a list of files that should chowned to
+		// the delegate uid in /sys/kernel/cgroup/delegate.  If the
+		// file is not present (Linux < 4.15), use the initial
+		// values mentioned in in cgroups(7).
+		filesToChown := []string{"."} // the directory itself must be chowned
+		cgroupDelegateFile := "/sys/kernel/cgroup/delegate"
```

Comment:
Ah! This is what I was talking about in https://github.com/opencontainers/runc/pull/3057/commits/56a353b13f93e60752ac52290df13e704db5d5d0#r700644360

Reply:
So maybe merge this one into the first commit

Reply:
Sure, I'm going to squash all commits.  Now that https://github.com/opencontainers/runtime-spec/pull/1123 was finally merged it is preferable to treat this whole PR as a single atomic commit.